### DISCUSSION TITLE
OpenAPI import limits

### DISCUSSION THREAD

**Discussion by kercsoj** [GitbookIO/community#442] [4. Open API] [answered]
Hi All,

  I wanted to explore the OpenAPI import functionality of Gitbook with the following sample OpenAPI file: https://gist.github.com/seriousme/55bd4c8ba2e598e416bb5543dcd362dc . Although the API spec is very small and it has just a few endpoints, Gitbook import fails with the message "_Some of your docs were not imported due to file limits, try splitting them into several imports._" I am pretty sure I didn't reach the 20 pages limit / import as specified here: https://docs.gitbook.com/content-editor/import#limitations

Can someone please tell me what I am doing wrong?

Thank you in advance!
  Jozsef Kercso

**Comment by aikoii** [comment]
👋 @kercsoj, I don't know if I'm missing anything, but I could import that file just fine. Are you using an [OpenAPI block ](https://docs.gitbook.com/content-editor/blocks/openapi)to import? 

<img width="754" alt="CleanShot 2024-01-05 at 20 12 51@2x" src="https://github.com/GitbookIO/community/assets/106068994/0cd6159d-eb7c-4607-b033-2cf7266cb8a5">

**Reply by kercsoj** [reply]
Hi @aikoii!

  thanks for your swift answer. No, I am not using the OpenAPI block, but using the "Import content" button. See recording about the issue: https://github.com/GitbookIO/community/assets/47670884/e0ca6275-9755-4a01-adce-c84ad9480c99

Thanks,
  Jozsef

**Reply by aikoii** [reply, marked as answer]
Thanks for sending over the recording, Jozsef!  "Import content" feature can be helpful, but it can be limiting. I'd recommend using OpenAPI block instead; it's the best way to incorporate APIs into GitBook🙂

**Reply by kercsoj** [reply]
Got it, thank you!

**Comment by minaelee** [comment]
I've experienced the same issue as originally reported. I also find (and suspect that it's true in OP's case as well, from their recording) that despite the error message, the OpenAPI spec actually DOES get imported fully, despite the error message. The error message appears to be incorrect.

**Reply by aikoii** [reply]
Hey @minaelee, thanks for sharing! Would it be possible to share the file so I can test it out?
